### DISCUSSION TITLE
Support Kotlin Flow as Subscription return type

### DISCUSSION THREAD

**Discussion by hantsy** [graphql-java/graphql-java#2853] [General]
When using Kotlin Coroutines in my project, I want to use Kotlin Coroutines API anywhere, it make my codes looks consistent, but it seems the Subscription does not support it.

```kotlin
val flow = MutableSharedFlow<Event>()

fun emitEvent() {
   flow.emit(event)
   ...
}

// subscription fun 
fun onEventTriggered() = flow
```

**Comment by bbakerman** [comment]
We had to choose a representation of a stream of subscribed events.  Being Java 8 based, we chose ReactiveStreams `Publisher` as our representation

You can write adaptors to make Kotlin flows turn into `Publishers`

As a rule we try to reduce of code dependencies towards zero (in fact if Java `Flow` was available to us we would have used that and NOT ReactiveStreams but alas it was not at the time)
